http-client/src/test/groovy/io/micronaut/http/client/StreamRequestSpec.groovy

```diff
@@ -88,8 +88,8 @@ class StreamRequestSpec extends Specification {
         )).contentType(MediaType.TEXT_PLAIN_TYPE), List)).blockFirst()
 
         then:
-        result.body().size() == 5
-        result.body() == ["Number 0", "Number 1", "Number 2", "Number 3", "Number 4"]
+        // no guarantee that the strings aren't merged, so we have to allow for it
+        result.body().join("") == "Number 0Number 1Number 2Number 3Number 4"
```

Comment:
can you explain this change?

Reply:
because it's TEXT_PLAIN mode, the client in this test simply sends the strings after one another, so the actual http stream is "Number 0Number 1Number 2Number 3Number 4". Each part is a different http chunk i think (often also a different tcp packet, but not always, which is why this test still worked sometimes).

With the change in this PR, the framework now can merge the different http chunks if they happen in the same TCP packet, which doesn't change the "HTTP stream", but changes how it's received by the controller (a single string vs a string for each part). 

it's not a problem if the user uses a format like json streams, where there is an actual delimiter between items